content/md/en/docs/reference/command-line-tools/node-template.md

```diff
@@ -147,86 +147,125 @@ Use the `node-template benchmark` command to analyze the resources required to e
-| `-r`, `--repeat <repeat>` | Specifies the number of times to repeat the execution of a benchmark from within the Wasm binary. The default is one execution.
+| `--record-proof` | Estimates the proof-of-validation (PoV) size.
+| `-r`, `--repeat <repeat>` | Specifies the number of times to repeat the execution of a benchmark from within the WebAssembly binary. The default is one execution.
 | `-s`, `--steps <steps>` | Specifies how many samples to take across the variable components. The default is one sample.
 | `--template <template>` | Specifies the path to a Handlebars template file used for outputting benchmark results.
-| `--tracing-receiver <receiver>` | Specifies the receiver to process tracing messages. The default is Log.
-| `--tracing-targets <targets>` | Sets a custom profiling filter. Syntax is the same as for logging: `<target>=<level>`.
-| `--wasm-execution <method>` | Specifies the method for executing Wasm runtime code. Valid values are `interpreted`, or `compiled`. The default is `Compiled`.
 
-#### Examples
+For examples of different benchmarking subcommands and the related command-line options, see [Benchmarking examples](#benchmarking-examples).
+
+#### Benchmarking examples
 
-To start a node and benchmark all of the extrinsic calls in all of the pallets, you might a command similar to the following:
+After you have compiled the runtime with benchmarking enabled, you can run a command similar to the following to benchmark all of the function calls in all of the pallets that have runtime-benchmarking configured:
 
 ```shell
-./target/release/node-template benchmark \
+./target/release/node-template benchmark pallet \
     --chain dev \
-    --execution wasm \
-    --wasm-execution compiled \
-    --extrinsic '\*' \
-    --pallet '\*' \
-    --output ./pallets/example/weights.rs
+    --extrinsic "*" \
+    --pallet "*" \
+    --output pallets/weights.rs
+```
+
+With this command, each function call is executed once with a single value and the resulting weight is recorded in the `weights.rs` file.
+
+Depending on the function you want to benchmark, you can add the `--steps` and `--repeat` command-line options to execute the call multiple times with different values.
+For example, the following command executes the `do_something` function in the `pallet_template` and calls the function 20 times to take 10 data points:
+
+```shell
+./target/release/node-template benchmark pallet \
+    --chain dev \
+    --extrinsic do_something \
+    --pallet pallet_template \
+    --steps 10 \
+    --repeat 20 \
+    --output pallets/weight2.rs
+```
+
+To measure the average, median, minimum, and maximum execution time per-block and per-extrinsic, you can run the `node-template benchmark overhead` subcommand:
+
+```shell
+./target/release/node-template benchmark overhead
+```
```

Comment:
If you are interested in adding some output here:  
```pre
Running 10 warmups...    
Executing block 100 times    
Per-block execution overhead [ns]:
	Total: 81779519
	Min: 695173, Max: 1522917
	Average: 817795, Median: 763360, Stddev: 160759.2
	Percentiles 99th, 95th, 75th: 1443343, 1263935, 814356    
	Writing weights to "block_weights.rs"    
Running 10 warmups...    
Executing block 100 times    
Building block, this takes some time...    
Extrinsics per block: 12000    
Running 10 warmups...    
Executing block 100 times
Per-extrinsic execution overhead [ns]:
	Total: 8955542
	Min: 85950, Max: 97448
	Average: 89555, Median: 88348, Stddev: 2713.05
	Percentiles 99th, 95th, 75th: 97247, 95040, 91733    
	Writing weights to "extrinsic_weights.rs"
```
Feel free to ignore if if you think if fits better at a different spot.